### PR TITLE
Problem: wc 2.0 not exposed to c++ bindings (fix #446)

### LINKED ISSUE #446: Problem: wc 2.0 not exposed to c++ bindings
expose  wc 2.0 functions to c++ bindings

### PR DESCRIPTION
**Pull request by leejw51crypto**
Solution: expose to c++
- new client
- restore client
- personal sign
- ensure session
- ping

### CODE REVIEW

File: demo/extra.cc [hunk 1]
```diff
@@ -163,6 +163,38 @@ Box<WalletconnectClient> make_new_client(std::string filename) {
     }
 }
 
+Box<Walletconnect2Client> make_new_client2(std::string filename) {
```
Comment (rarcifa) [inline demo/extra.cc:173]: Maybe we can add some documentation here as well as for other functions? e.g. 
/**
 * @summary 
 *
 * @param
 * @return 
 * @throws 
 */

File: demo/extra.cc [hunk 2]
```diff
@@ -163,6 +163,38 @@ Box<WalletconnectClient> make_new_client(std::string filename) {
...
+        String client_meta_json =
+            "{\"description\":\"Defi WalletConnect v2 "
+            "example.\",\"url\":\"http://localhost:8080/"
+            "\",\"icons\":[],\"name\":\"Defi WalletConnect Web3 Example\"}";
+
+        String projectid = getEnv("NEXT_PUBLIC_PROJECT_ID");
+        Box<Walletconnect2Client> client = walletconnect2_client_new(
+            relay_server_string.c_str(), projectid.c_str(),
+            required_namespaces_json.c_str(), client_meta_json.c_str());
+        cout << "qrcode= " << client->get_connection_string() << endl;
```
Comment (rarcifa) [inline demo/extra.cc]: std::cout ?

File: demo/extra.cc [hunk 3]
```diff
@@ -317,6 +348,72 @@ void test_wallet_connect() {
     }
 }
 
+void test_wallet_connect2() {
```
Comment (rarcifa) [inline demo/extra.cc:367]: here as well

**Review by rarcifa — APPROVED** [review]
approved with comments

File: demo/extra.cc (hunk 1 above)
Comment (leejw51crypto) [reply demo/extra.cc:173]: ok

File: demo/extra.cc (hunk 2 above)
Comment (leejw51crypto) [reply demo/extra.cc]: ok

File: demo/extra.cc (hunk 3 above)
Comment (leejw51crypto) [reply demo/extra.cc:367]: ok, i'll add comment

**Pull request merged**